spyder_kernels/customize/spyderpdb.py

```diff
@@ -703,7 +704,13 @@ def cmd_input(self, prompt=''):
         if is_main_thread and kernel.eventloop:
             while self._cmd_input_line is None:
                 eventloop = kernel.eventloop
-                if eventloop:
+                # Check if the current backend is Tk on Windows
+                # to let GUI update.
+                # See spyder-ide/spyder#17523
+                if (eventloop and hasattr(kernel, "app_wrapper") and
+                        os.name == "nt"):
```

Comment:
Did you check that this is not necessary for Windows or Mac?

Reply:
Checked on Windows and Linux. On Windows the change is necessary on Linux no. I think the reason is because on Linux/Mac the tk eventloop is used with the `createfilehandler` method which is not available for Windows. Should I test on Mac too?

Reply:
Yeah, please do. I meant Linux and Mac above (sorry for the confusion).

Reply:
Checked and seems like is not necessary with Mac (so the change is only necessary for Windows) @ccordoba12

Reply:
Ok, great! Merging then.